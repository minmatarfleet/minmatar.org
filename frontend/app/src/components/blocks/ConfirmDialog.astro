---
import { i18n } from '@helpers/i18n'
const { t } = i18n(Astro.url)

import Dialog from '@components/blocks/Dialog.astro';
import Flexblock from '@components/compositions/Flexblock.astro';
import FlexInline from '@components/compositions/FlexInline.astro';
import Button from '@components/blocks/Button.astro';

interface Props {
    [propName: string]: any;
}
---

<div
    class="[ confirm-dialog ]"
>
    <Dialog
        x-data={`{
            close_dialog() {
                if (!confirm_dialog_open)
                    return

                var clickEvent = new MouseEvent("click", {
                    "view": window,
                    "bubbles": true,
                    "cancelable": false
                });

                document.querySelector('.confirm-dialog-backdrop').dispatchEvent(clickEvent);
            }
        }`}
        @keyup.escape.window="close_dialog()"
        x-bind:class="(confirm_dialog_open ? 'open' : '')"
        x-bind:aria-hidden="confirm_dialog_open == false"
        breakout={true}
        class="[ w-full max-w-[500px] max-h-[80vh] overflow-y-auto ]"
        aria-labelledby="confirm-dialog-title"
        aria-describedby="confirm-dialog-description"
        x-trap="confirm_dialog_open"
        x-bind:inert="!confirm_dialog_open"
    >
        <Flexblock gap="var(--space-l)">
            <Flexblock gap="var(--space-s)">
                <h3 id="confirm-dialog-title" x-text="confirm_dialog_title"></h3>
                <div
                    id="confirm-dialog-description"
                    class="[ confirm-dialog-content ]"
                    x-html="confirm_dialog_content"
                    x-bind:hx-get="confirm_dialog_partial ? (confirm_dialog_partial ?? false) : false"
                    x-bind:hx-trigger="confirm_dialog_partial ? 'intersect' : false"
                    x-bind:hx-swap="confirm_dialog_partial ? 'innerHTML transition:true' : false"
                    hx-target="#confirm-dialog-description"
                    hx-indicator=".loader"
                ></div>
            </Flexblock>
                
            <FlexInline justification='flex-end'>
                <Button
                    size='sm'
                    confirm-dialog-action="accept"
                    x-bind:hx-get="confirm_dialog_hx?.method == 'get' ? (confirm_dialog_hx?.url ?? false) : false"
                    x-bind:hx-post="confirm_dialog_hx?.method == 'post' ? (confirm_dialog_hx?.url ?? false) : false"
                    x-bind:hx-put="confirm_dialog_hx?.method == 'put' ? (confirm_dialog_hx?.url ?? false) : false"
                    x-bind:hx-patch="confirm_dialog_hx?.method == 'patch' ? (confirm_dialog_hx?.url ?? false) : false"
                    x-bind:hx-delete="confirm_dialog_hx?.method == 'delete' ? (confirm_dialog_hx?.url ?? false) : false"
                    x-bind:hx-target="confirm_dialog_hx?.target ?? false"
                    x-bind:hx-swap="confirm_dialog_hx?.swap ?? 'none'"
                    x-bind:hx-vals="confirm_dialog_hx?.vals ? JSON.stringify({ data: JSON.stringify(confirm_dialog_hx?.vals) }) : false"
                    hx-indicator=".loader"
                    x-bind:disabled="confirm_dialog_accept_disabled"
                >
                    {t('yes')}
                </Button>
                <Button size='sm' confirm-dialog-action="cancel">{t('no')}</Button>
            </FlexInline>
        </Flexblock>
    </Dialog>

    <div
        class="confirm-dialog-backdrop"
        x-bind:class="{ open: confirm_dialog_open }"
        confirm-dialog-action="cancel"
    />
</div>

<style lang="scss">
    :global([confirm-dialog-action] *) {
        pointer-events: none !important;
    }

    :global(.confirm-dialog .confirm-dialog-content p) {
        font-size: var(--step--1);
    }

    .confirm-dialog-backdrop {
        position: fixed;
        inset: 0;
        z-index: var(--backdrop-z-index);
        display: none;

        &.open {
            display: block;
        }
    }
</style>