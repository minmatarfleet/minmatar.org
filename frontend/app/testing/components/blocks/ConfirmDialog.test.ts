import { experimental_AstroContainer as AstroContainer } from "astro/container";
import { expect, test, vi } from "vitest";
import ConfirmDialog from "@components/blocks/ConfirmDialog.astro";

test("ConfirmDialog defaults", async () => {
  const container = await AstroContainer.create();
  const result = await container.renderToString(ConfirmDialog, {
    slots: {
      default: "slotted chick",
    },
  });

  // expect(result).toMatchSnapshot();
});
