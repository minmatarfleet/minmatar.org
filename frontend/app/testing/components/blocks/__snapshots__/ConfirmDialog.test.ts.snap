// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ConfirmDialog defaults 1`] = `
"<div class="[ confirm-dialog ]" data-astro-cid-kediyatg> <div role="dialog" class="dialog solid-transparency breakout fixed [ w-full max-w-[500px] max-h-[80vh] overflow-y-auto ]" x-data="{
            close_dialog() {
                if (!confirm_dialog_open)
                    return

                var clickEvent = new MouseEvent(&#34;click&#34;, {
                    &#34;view&#34;: window,
                    &#34;bubbles&#34;: true,
                    &#34;cancelable&#34;: false
                });

                document.querySelector('.confirm-dialog-backdrop').dispatchEvent(clickEvent);
            }
        }" @keyup.escape.window="close_dialog()" x-bind:class="(confirm_dialog_open ? 'open' : '')" x-bind:aria-hidden="confirm_dialog_open == false" aria-labelledby="confirm-dialog-title" aria-describedby="confirm-dialog-description" x-trap="confirm_dialog_open" x-bind:inert="!confirm_dialog_open" data-astro-cid-kediyatg="true"> <div class="wrapper" data-mod data-astro-cid-yv35cmqx style="--wrapper_padding_inline: var(--space-m);--wrapper_padding_inline_end: var(--space-m);--wrapper_padding_block: var(--space-m);--wrapper_max_width: 80rem;">  <div class="flex-block gap---space-m" data-mod style="--flex_block_gap: var(--space-m);">   <div class="[ cover-body ]">  <div class="flex-block gap---space-l" data-astro-cid-kediyatg="true" data-mod style="--flex_block_gap: var(--space-l);">  <div class="flex-block gap---space-s" data-astro-cid-kediyatg="true" data-mod style="--flex_block_gap: var(--space-s);">  <h3 id="confirm-dialog-title" x-text="confirm_dialog_title" data-astro-cid-kediyatg></h3> <div id="confirm-dialog-description" class="[ confirm-dialog-content ]" x-html="confirm_dialog_content" x-bind:hx-get="confirm_dialog_partial ? (confirm_dialog_partial ?? false) : false" x-bind:hx-trigger="confirm_dialog_partial ? 'intersect' : false" x-bind:hx-swap="confirm_dialog_partial ? 'innerHTML transition:true' : false" hx-target="#confirm-dialog-description" hx-indicator=".loader" data-astro-cid-kediyatg></div>  </div>  <div class="flex-inline" data-astro-cid-kediyatg="true" data-astro-cid-ud5fqf5k style="--flex_inline_gap: var(--space-s);--flex_inline_justification: flex-end;">  <button confirm-dialog-action="accept" x-bind:hx-get="confirm_dialog_hx?.method == 'get' ? (confirm_dialog_hx?.url ?? false) : false" x-bind:hx-post="confirm_dialog_hx?.method == 'post' ? (confirm_dialog_hx?.url ?? false) : false" x-bind:hx-put="confirm_dialog_hx?.method == 'put' ? (confirm_dialog_hx?.url ?? false) : false" x-bind:hx-patch="confirm_dialog_hx?.method == 'patch' ? (confirm_dialog_hx?.url ?? false) : false" x-bind:hx-delete="confirm_dialog_hx?.method == 'delete' ? (confirm_dialog_hx?.url ?? false) : false" x-bind:hx-target="confirm_dialog_hx?.target ?? false" x-bind:hx-swap="confirm_dialog_hx?.swap ?? 'none'" x-bind:hx-vals="confirm_dialog_hx?.vals ? JSON.stringify({ data: JSON.stringify(confirm_dialog_hx?.vals) }) : false" hx-indicator=".loader" x-bind:disabled="confirm_dialog_accept_disabled" data-astro-cid-kediyatg="true" data-size="sm" style="--button_background_color: var(--fleet-red); --button_color: var(--fleet-yellow)" disabled x-init="$el.removeAttribute(&#34;disabled&#34;)" data-astro-cid-r5e2wo56="true" class="[ button ]"> <div class="flex-inline" data-astro-cid-r5e2wo56="true" data-astro-cid-ud5fqf5k style="--flex_inline_gap: var(--space-xs);--flex_inline_justification: center;">   Yes  </div>   </button>  <button confirm-dialog-action="cancel" data-astro-cid-kediyatg="true" data-size="sm" style="--button_background_color: var(--fleet-red); --button_color: var(--fleet-yellow)" disabled x-init="$el.removeAttribute(&#34;disabled&#34;)" data-astro-cid-r5e2wo56="true" class="[ button ]"> <div class="flex-inline" data-astro-cid-r5e2wo56="true" data-astro-cid-ud5fqf5k style="--flex_inline_gap: var(--space-xs);--flex_inline_justification: center;">   No  </div>   </button>   </div>    </div>   </div>   </div>   </div>   </div>  <div class="confirm-dialog-backdrop" x-bind:class="{ open: confirm_dialog_open }" confirm-dialog-action="cancel" data-astro-cid-kediyatg></div> </div> "
`;
